tnt/src/main/java/org/dcsa/conformance/standards/tnt/party/TntPublisher.java

```diff
@@ -112,13 +130,167 @@ private void supplyScenarioParameters(JsonNode actionPrompt) {
   public ConformanceResponse handleRequest(ConformanceRequest request) {
     log.info("TntPublisher.handleRequest(%s)".formatted(request));
 
-    JsonNode jsonResponseBody =
-        JsonToolkit.templateFileToJsonNode(
-            "/standards/tnt/messages/tnt-220-response.json", Map.ofEntries());
+    Map<String, List<AttributeMapping>> attributeMappings = AttributeMapping.initializeAttributeMappings();
+
+    ObjectMapper mapper = new ObjectMapper();
+    ArrayNode filteredArray = mapper.createArrayNode();
+
+    ArrayNode jsonResponseBody = (ArrayNode) JsonToolkit.templateFileToJsonNode(
+        "/standards/tnt/messages/tnt-220-response.json", Map.ofEntries());
 
+    if (!new QueryParameterSpecificRule().validate(request.queryParams())) {
+      return return400(request,"Error: Invalid query parameters provided.");
+    }
+
+    Set<String> eventIds = new HashSet<>();
+
+    for (Map.Entry<String, ? extends Collection<String>> queryParam : request.queryParams().entrySet()) {
+      String paramName = queryParam.getKey();
+      Collection<String> paramValues = queryParam.getValue();
+      List<AttributeMapping> mappings = attributeMappings.get(paramName);
+      if (mappings != null) {
+        for (AttributeMapping mapping : mappings) {
+          for (String paramValue : paramValues) {
+            String jsonPath = mapping.getJsonPath();
+            BiPredicate<JsonNode,String> condition = mapping.getCondition();
+            for (JsonNode node : jsonResponseBody) {
+              List<JsonNode> results = new ArrayList<>();
+              traverse(node, jsonPath.split("/"), 0, results, condition, paramValue);
+              if(!results.isEmpty()){
+              var eventId = node.at("/eventID").asText();
+              if ( !eventIds.contains(eventId)
+                && (mapping.getValues().isEmpty()
+                  || results.stream().anyMatch(result -> mapping.getValues().contains(result.asText())))) {
+                  eventIds.add(eventId);
+                  filteredArray.add(node);
+                }
+              }
+            }
+          }
+        }
+      }
+    }
```

Comment:
This whole code block above (line 147 until 172) is quite complex and hard to understand what is going on. Can yo u e.g. extract this as a method? That already helps.

By the way, is this logic quite generic? To filter out parameters from a `ConformanceRequest`? Should we have/use (add?) something in core module, to do this?

Reply:
done